test/libyul/yulOptimizerTests/reasoningBasedSimplifier/mulmod.yul

```diff
@@ -0,0 +1,31 @@
+{
+    let x := calldataload(0)
+    let y := calldataload(32)
+    let z := calldataload(64)
+    let result := mulmod(x, y, z)
+
+    // should be zero
+    if gt(result, z) { sstore(0, 1) }
+
+    // mulmod is equal to mod of product for small numbers
+    if and(and(lt(x, 1000), lt(y, 1000)), lt(z, 1000)) {
+        if eq(result, mod(mul(x, y), z)) { sstore(0, 9) }
+    }
+
+    // but not in general
+    if and(and(gt(x, sub(0, 5)), gt(y, sub(0, 2))), eq(z, 3)) {
+        if eq(result, mod(mul(x, y), z)) { sstore(0, 5) }
```

Comment:
I'm not sure that's what's happening, but the archlinux CI test run failing to simplify this one, may indicate that we are already hitting Z3 determinism issues here (i.e. 4.8.8 might solve this, but 4.8.9 might not) - but maybe something else is going on...

Reply:
let me try to reduce the range of the numbers.

Reply:
it's not only arch though, the ubuntu ones are too

Reply:
and those are still on 4.8.8

Reply:
It would be nice and fancy, if we ran these tests as semantics, resp. Yul interpreter tests as well, and compared traces of optimized and unoptimized versions :-)... Then we could be sure or at least more confident that the SMT solver does the right thing and the SMT encoding is correct... but nothing for this PR, but we could consider it in general.